clients/src/main/java/org/apache/kafka/common/telemetry/ClientTelemetryReceiver.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.telemetry;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.server.authorizer.AuthorizableRequestContext;
+
+@InterfaceStability.Evolving
+public interface ClientTelemetryReceiver {
```

Comment:
Originally this interface was implemented and executed on the broker upon receipt of the payload. Is it now also optionally executed on the client?

Reply:
It ll be for broker only.

Reply:
If it's for broker only, it should no be added to `clients` module, right?

Reply:
Make sense, shall I move it to `server-common/org/apache/kafka/metrics/`? Is there any other better place for these i.e ClienTelemetryReceiver.java, ClientTelemetry.java and ClientTelemetryPayload.java

Reply:
Thinking again I think it should be good to layout Client Specific Interfaces which broker might implement if only wants to support client telemetry in client packages. We shouldn't bloat server packages with interfaces which an external dependency needs to implement. Please let me know your thoughts.